src/umpire/strategy/DynamicPoolList.cpp

```diff
@@ -125,20 +125,28 @@ void DynamicPoolList::coalesce() noexcept
   umpire::event::record([&](auto& event) {
     event.name("coalesce").category(event::category::operation).tag("allocator_name", getName()).tag("replay", "true");
   });
-  dpa.coalesce(dpa.getActualSize());
+
+  std::size_t suggested_size{m_should_coalesce(*this)};
+  if (0 != suggested_size) {
+    UMPIRE_LOG(Debug,
+               "Heuristic returned true, "
+               "performing coalesce operation for "
+                   << this << "\n");
```

Comment:
What does the output of this debug log look like?  You can enable it by running with `UMPIRE_LOG_LEVEL=Debug` and then look at the debug file output for this particular `UMPIRE_LOG`.

Reply:
There was no output for that part of the debug log.

Reply:
You should see some if that code gets executed and a coalesce happening.  I'm concerned because I do not know what is expected with `<< this <<` in the input.  What is is that you are trying to log here?  Is it necessarily?  If so, think you need to change it somewhat to display the information that will be useful for debugging.  

Also, if you do decide to keep it, we also need to change `"\n"' to `std::endl`

You might want to work with @kab163 on this because you should being seeing output in the file when this code is executed and the environment variable is set.